tests/josegonzalez/Queuesadilla/Engine/AbstractPdoEngineTest.php

```diff
@@ -0,0 +1,246 @@
+<?php
+namespace josegonzalez\Queuesadilla\Engine;
+
+use josegonzalez\Queuesadilla\FixtureData;
+use josegonzalez\Queuesadilla\TestCase;
+use PDO;
+use PDOException;
+use Psr\Log\NullLogger;
+
+abstract class AbstractPdoEngineTest extends TestCase
+{
+
+    /**
+     * Sets up the test case. Sub classes need to set the $url and $engineClass
+     * properties before calling parent::setUp()
+     *
+     * @return void
+     */
+    public function setUp()
+    {
+        $this->config = ['url' => $this->url];
+        $this->Logger = new NullLogger;
+        $this->Engine = $this->mockEngine();
+        $this->Fixtures = new FixtureData;
+        $this->clearEngine();
+    }
+
+    /**
+     * Used to truncate the jobs table and to reset the auto increment value.
+     *
+     * @return void
+     */
+    abstract protected function clearEngine();
+
+    /**
+     * Tear Down
+     *
+     * @return void
+     */
+    public function tearDown()
+    {
+        $this->clearEngine();
+        unset($this->Engine);
+    }
+
+    /**
+     * @covers josegonzalez\Queuesadilla\Engine\PdoEngine::__construct
+     */
+    public function testConstruct()
+    {
+        if ($this->Engine->connection() === null) {
+            $this->markTestSkipped('No connection to database available');
+        }
+        $Engine = new $this->engineClass($this->Logger, [
+            'user' => 'invalid'
+        ]);
+        $this->assertNull($Engine->connection());
+
+        $Engine = new $this->engineClass($this->Logger, $this->url);
+        $this->assertNotNull($Engine->connection());
+
+        $Engine = new $this->engineClass($this->Logger, $this->config);
+        $this->assertNotNull($Engine->connection());
+    }
+
+    /**
+     * @covers josegonzalez\Queuesadilla\Engine\PdoEngine::connect
```

Comment:
I think the reason test coverage went down is that the MysqlEngine is no longer covered here.